### ISSUE TITLE
[Dropdown] options are not keyboard accessible

### ISSUE THREAD

**Issue by akdetrick** [narmi/design_system#573]
### Component/Module
`Dropdown`

### Expected Result
You should be able to navigate options with arrow keys

### Actual Result
You must use a mouse to select an option

### Additional Details / Context
A hook from the `downshift` library might be helpful here.

**Comment by akdetrick** [comment]
will be handled in #630
